src/meds_etl/simple_csv.py

```diff
@@ -0,0 +1,220 @@
+import argparse
+import json
+import multiprocessing
+import os
+import pickle
+import random
+import shutil
+import subprocess
+import tempfile
+
+import jsonschema
+import meds
+import polars as pl
+import pyarrow.parquet as pq
+
+import meds_etl
+
+
+def load_file(decompressed_dir, fname):
+    if fname.endswith(".gz"):
+        file = tempfile.NamedTemporaryFile(dir=decompressed_dir)
+        subprocess.run(["gunzip", "-c", fname], stdout=file)
+        return file
+    else:
+        return open(fname)
+
+
+def process_file(args):
+    event_file, num_shards, temp_dir, decompressed_dir, index = args
+    print("Working on ", event_file)
+
+    with load_file(decompressed_dir, event_file) as temp_f:
+        table = pl.read_csv_batched(temp_f.name, infer_schema_length=0, batch_size=1_000_000)
+
+        batch_index = 0
+
+        while True:
+            batch_index += 1
+
+            batch = table.next_batches(1)
+            if batch is None:
+                break
+
+            batch = batch[0]
+
+            batch = batch.lazy().rename({c: c.lower() for c in batch.columns})
+
+            patient_id = pl.col("patient_id").cast(pl.Int64)
+            time = pl.col("time")
+            
+            
+            time = pl.coalesce(
+                time.str.to_datetime("%Y-%m-%d %H:%M:%S%.f", strict=False),
+                time.str.to_datetime("%Y-%m-%d", strict=False),
+            )
+
+            value = pl.col("value")
+            code = pl.col("code")
+
+            datetime_value = pl.coalesce(
```

Comment:
I'm not sure this is the right strategy. I think this also gets addressed a bit if we move away from CSV, but with this strategy you risk having things being assigned to the wrong column just because they can be converted to that type. E.g., if I have an ordinal scale that reads "<1, 1, 2, >2" as all strings, then half of the scale will end up as numerics and the other half as text, which I think is sub-optimal. Better to just have the input format have all three columns and let the user dictate what goes where.

Reply:
Yeah, this is a good point. Explicit is better than implicit. I'll fix this and then resolve this comment.